apps/jubmoji-quest/src/hooks/useFetchQuests.ts

```diff
@@ -1,5 +1,41 @@
 import { JubmojiQuest } from "@/types";
 import { useQuery } from "react-query";
+import { useJubmojis } from "./useJubmojis";
+
+export const useGetQuestPowerLockedStatus = (questId?: string | number) => {
+  const { data: jubmojis = [] } = useJubmojis();
+  const { data: quest } = useFetchQuestById(`${questId}`);
+
+  return useQuery(
+    ["questPowerLockedStatus", questId, jubmojis.length],
+    async (): Promise<{ locked: boolean }> => {
+      if (!quest) {
+        return {
+          locked: true,
+        };
+      }
+
+      const collectionCardIndices = quest.collectionCards.map(
+        (card) => card.index
+      );
+
+      const collectedItems =
+        jubmojis?.filter((jubmoji) =>
+          collectionCardIndices.includes(jubmoji.pubKeyIndex)
+        ).length ?? 0;
+
+      const collectionTotalItems = quest.collectionCards.length;
+
+      const powerIsLocked = collectedItems < collectionTotalItems;
```

Comment:
I think we might want to change this - for most powers/quests you can use them as long as you have one card. I think the `powerIsLocked` logic actually will depend on `quest.ProofType`

Reply:
Maybe for now, its unlocked as long as you've collected one of the cards - and I can update this logic in a separate change